File: OBC/scheduling/state_processes/state_bootup.py
# ///////////////////////////////////////////////////////////////// #
# BOOTUP STATE
# ///////////////////////////////////////////////////////////////// #


# IMPORTS
import time


# FUNCTIONS
def __init__():
    on_loop()


def on_loop():
    # TO-DO: do some setup functions
    # ....
    time.sleep(1) 


# MAIN FUNCTION
print("\n \033[38;5;46m[STATE_BOOTUP] [Started]\033[0m \n", flush=True)
__init__()
print("\n \033[38;5;196m[STATE_BOOTUP] [Ended]\033[0m \n", flush=True)

File: OBC_C++/scheduling/state_processes/state_bootup.cpp
// state_bootup.py


// ++++++++++++++ Imports/Installs ++++++++++++++ //
#include <iostream>
#include <thread>
#include <chrono>


// ++++++++++++++ Helper Functions ++++++++++++++ //
void on_loop() {
    // TO-DO: do some setup functions
    // ....
    std::this_thread::sleep_for(std::chrono::seconds(5)); // Sleep for 5 seconds
}

void __init__() {
    on_loop();
}


// ++++++++++++++ Main Function ++++++++++++++ //
int main() {
    std::cout << "\n\033[38;5;46m[STATE_BOOTUP] [Started]\033[0m\n" << std::flush;
    __init__();
    std::cout << "\n\033[38;5;196m[STATE_BOOTUP] [Ended]\033[0m\n" << std::flush;
    return 0;
}
